Stripe/STPCustomer.m

```diff
@@ -85,43 +83,60 @@ + (instancetype)decodedObjectFromAPIResponse:(NSDictionary *)response {
         shipping.country = shippingDict[@"address"][@"country"];
         customer.shippingAddress = shipping;
     }
-    NSMutableArray *sources = [NSMutableArray array];
-    if ([dict[@"sources"] isKindOfClass:[NSDictionary class]] && [dict[@"sources"][@"data"] isKindOfClass:[NSArray class]]) {
-        for (id contents in dict[@"sources"][@"data"]) {
-            if ([contents isKindOfClass:[NSDictionary class]]) {
-                if ([contents[@"object"] isEqualToString:@"card"]) {
-                    STPCard *card = [STPCard decodedObjectFromAPIResponse:contents];
-                    // ignore apple pay cards from the response
-                    if (card && !card.isApplePayCard) {
-                        [sources addObject:card];
-                        if (defaultSourceId && [card.stripeID isEqualToString:defaultSourceId]) {
-                            customer.defaultSource = card;
-                        }
+    [customer updateSourcesWithResponse:dict filteringApplePay:YES];
+    customer.allResponseFields = dict;
+    return customer;
+}
+
+- (void)updateSourcesWithResponse:(NSDictionary *)response
+                filteringApplePay:(BOOL)filterApplePay {
+    NSArray *data;
+    if ([response[@"sources"] isKindOfClass:[NSDictionary class]]) {
+        data = response[@"sources"][@"data"];
+    }
+    if (![data isKindOfClass:[NSArray class]]) {
+        return;
+    }
+    NSString *defaultSourceId;
```

Comment:
I think you should probably reset `self.defaultSource = nil`.

ex: calling with `filterApplePay = NO` and then with `YES`, for a customer whose default source is ApplePay, will still have the apple pay `defaultSource` on the customer object.

Reply:
ah yep, good catch!